### PR TITLE
Link to bash in env reporting

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

This PR adds some more information to `accelerate env` to help with knowing *what* `accelerate launch` is being ran, to help solve issues like https://github.com/huggingface/accelerate/issues/2622 faster

For example, here is the new output:

```
Copy-and-paste the text below in your GitHub issue

- `Accelerate` version: 0.29.0.dev0
- Platform: Linux-6.5.0-25-generic-x86_64-with-glibc2.35
- `accelerate` bash location: /home/zach/miniconda3/envs/accelerate/bin/accelerate
- Python version: 3.9.15
- Numpy version: 1.24.0
- PyTorch version (GPU?): 2.2.1+cu121 (True)
- PyTorch XPU available: False
- PyTorch NPU available: False
- PyTorch MLU available: False
- System RAM: 62.50 GB
- GPU type: NVIDIA GeForce RTX 3070 Laptop GPU
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: NO
        - mixed_precision: no
        - use_cpu: False
        - debug: False
        - num_processes: 1
        - machine_rank: 0
        - num_machines: 1
        - rdzv_backend: static
        - same_network: False
        - main_training_function: main
        - enable_cpu_affinity: False
        - downcast_bf16: False
        - tpu_use_cluster: False
        - tpu_use_sudo: False
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@SunMarc @BenjaminBossan

### CODE REVIEW

File: src/accelerate/commands/env.py [hunk 1]
```diff
@@ -55,9 +56,15 @@ def env_command(args):
     if args.config_file is not None or os.path.isfile(default_config_file):
         accelerate_config = load_config_from_file(args.config_file).to_dict()
 
+    # if we can run which, get it
+    try:
+        bash_location = subprocess.check_output(["which", "accelerate"], text=True, stderr=subprocess.STDOUT).strip()
```
Comment (BenjaminBossan) [inline src/accelerate/commands/env.py:61]: Does this work on Windows?

Comment (muellerzr) [reply src/accelerate/commands/env.py:61]: It does not, hence the check here.

Comment (muellerzr) [reply src/accelerate/commands/env.py:61]: Looks like `where` is the equivalent

Comment (muellerzr) [reply src/accelerate/commands/env.py]: Added windows equivalent checks

**Pull request merged**